azure-pipelines.yaml

```diff
@@ -18,40 +18,38 @@ stages:
         steps:
 
           - task: AzureCLI@2
-            name: GetCertificateFromKeyVault
+            name: GetCertificateFromKeyVaultAndDecide
             inputs:
-              azureSubscription: $(KEY_VAULT_SERVICE_CONNECTION)
+              azureSubscription: '$(KEY_VAULT_SERVICE_CONNECTION)'
               scriptType: bash
               scriptLocation: inlineScript
               inlineScript: |
-                az keyvault certificate download --vault-name '$(KEY_VAULT_NAME)' -n '$(KEY_VAULT_CERT_NAME)' -f certificate.pem >/dev/null 2>&1
-            condition: eq(${{parameters.force_renew_cert}}, False)
-
-          - script: pip3 install --require-hashes --requirement requirements.txt
-            name: InstallPythonDependencies
-            condition: and(succeeded(), eq(${{parameters.force_renew_cert}}, False))
-
-          - task: PythonScript@0
-            name: CheckCertificateExpiration
-            condition: and(succeeded(), eq(${{parameters.force_renew_cert}}, False))
-            inputs:
-              scriptSource: filePath
-              scriptPath: check_certificate_expiry.py
-              arguments: --certificate certificate.pem --delta $(CERT_NAME_EXPIRE_SECONDS)
+                if [ ${{ parameters.force_renew_cert }} == "True" ]
+                then
+                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
+                  exit 0
+                fi
+                if ! az keyvault certificate download --vault-name '$(KEY_VAULT_NAME)' -n '$(KEY_VAULT_CERT_NAME)' -f certificate.pem >/dev/null
+                then
+                  echo "##vso[task.setvariable variable=require_new_certificate;isOutput=true]True"
+                  exit 0
+                else
+                  if ! openssl x509 -checkend '$(CERT_NAME_EXPIRE_SECONDS)' -noout -in certificate.pem >/dev/null 2>&1
+                  then
+                    # certificate is expiring or is already expired
```

Comment:
```suggestion
                   echo "certificate is expiring or is already expired"
```

Reply:
Commit 4d516a4 removes /dev/null output from openssl, making it print this info.